// @ngx-loading-bar/http-client/esm2022/loading-bar.interceptor.mjs
import { Injectable } from '@angular/core';
import { HttpContextToken } from '@angular/common/http';
import { finalize, tap } from 'rxjs/operators';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-loading-bar/core";
export const NGX_LOADING_BAR_IGNORED = new HttpContextToken(() => false);
export class LoadingBarInterceptor {
    constructor(loader) {
        this.loader = loader;
    }
    intercept(req, next) {
        if (req.context.get(NGX_LOADING_BAR_IGNORED) === true) {
            return next.handle(req);
        }
        let started = false;
        const ref = this.loader.useRef('http');
        return next.handle(req).pipe(tap(() => {
            if (!started) {
                ref.start();
                started = true;
            }
        }), finalize(() => started && ref.complete()));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LoadingBarInterceptor, deps: [{ token: i1.LoadingBarService }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LoadingBarInterceptor }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.12", ngImport: i0, type: LoadingBarInterceptor, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.LoadingBarService }]; } });
export function provideLoadingBarInterceptor() {
    return {
        provide: HTTP_INTERCEPTORS,
        useClass: LoadingBarInterceptor,
        multi: true,
    };
}
